### PR TITLE
manual: add chapter on Grsecurity/PaX

### PR DESCRIPTION
**Pull request by joachifm**
Document configuration and administration of Grsecurity/PaX on NixOS systems. I hope this helps explain the "what", "why", and "how" of this feature.

I'd like critcism of precision, clarity, and technical correctness, but any notes are very much appreciated.

### CODE REVIEW

**Comment by joachifm** [comment]
cc  @copumpkin @thoughtpolice

**Comment by copumpkin** [comment]
Sweet, thanks! I'll check it out in the next day or so

**Comment by joachifm** [comment]
I should also mention that I'm a docbook novice, so there's probably lots to be desired on that front as well.

File: nixos/doc/manual/configuration/grsecurity.xml [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+
+  </para>
+
+  </section>
+
+  <section><title>Using a custom Grsecurity/PaX kernel</title>
+
+  <para>
+
+    The NixOS kernel is suitable for causal desktop users, but for specialized
```
Comment (fpletz) [inline nixos/doc/manual/configuration/grsecurity.xml]: s/causal/casual/

**Comment by teh** [comment]
I wonder whether we should add a comment here about nixos following the dev branches only (due to grsecurity's closed stable approach). I have a grsecurity kernel and I need to re-compile every few weeks.

**Comment by joachifm** [comment]
@teh makes sense, I'll work it into the next revision. From my pov, the users I wish to cater to shouldn't have to care about such details, but I can see how being committed to running latest linux -stable can be a problem for some people, so it's good to be up front about it.

File: nixos/doc/manual/configuration/grsecurity.xml [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+    ]]></screen>
+  </para>
+
+  </sect1>
+
+  <sect1 xml:id="sec-grsec-declarative-tuning"><title>Declarative tuning</title>
+
+  <para>
+    The NixOS module exposes only a limited number of options for tuning the
+    beahvior of grsecurity/PaX.  These are options thought to be of particular
```
Comment (danbst) [inline nixos/doc/manual/configuration/grsecurity.xml]: `beahvior` typo

File: nixos/doc/manual/configuration/grsecurity.xml [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+          kernel = pkgs.linux_grsec_nixos.override {
+            extraConfig = ''
+              GRKERNSEC y
+              PAX y
+              GRKERNSEC_CONFIG_AUTO y
+              GRKERNSEC_CONFIG_SERVER y
+              GRKERNSEC_CONFIG_SECURITY y
+            '';
+          };
+          self = pkgs.linuxPackagesFor kern self;
```
Comment (danbst) [inline nixos/doc/manual/configuration/grsecurity.xml]: You reference `kern`, but it is not defined. Maybe typo `kernel`?

**Comment by danbst** [comment]
There is a note about this as desktop feature, but never explained why not server feature. I remember you've explained it somewhere, maybe include that explanation here?

File: nixos/doc/manual/configuration/grsecurity.xml (hunk 3 above)
Comment (joachifm) [inline nixos/doc/manual/configuration/grsecurity.xml]: Yep, that's a typo. Thank you

**Comment by joachifm** [comment]
@danbst it's basically to avoid combinatorial explosion, standardising on a single package set improves the feasibility of testing.

**Comment by joachifm** [comment]
I'm reasonably happy with this now. Thanks for taking a look.

**Pull request merged**